Update the documentation to match what driver.sh actually does now (#273)
ClangBuiltLinux/continuous-integration: usage.txt

  AR:
      If no AR value is specified, the script will attempt to set AR to

>       llvm-ar, then ${CROSS_COMPILE}ar if they are found in PATH.

Reviewer: ```suggestion
      llvm-ar, then ${CROSS_COMPILE}ar if it is found in PATH.
```

>       llvm-ar, then ${CROSS_COMPILE}ar if they are found in PATH.

Reviewer 2: I might word all of these as "the script will attempt to see <VAR> to <llvm_name> and fallback to ${CROSS_COMPILE}<gnu_name> if it is not found in PATH".

I know Nick was wanting to eliminate that behavior but still.

Otherwise, Harsh's changes also seem good to me.

Author: Actually this makes more sense than current loops in driver.sh, so maybe the logic there should be revisited.

  ARCH (required):
      If no ARCH value is specified, it's an error; there is no default.
      Currently, arm32_v7, arm32_v6, arm32_v5, arm64, mips, mipsel, ppc32,
      ppc64, ppc64le, s390, riscv, x86, and x86_64 are supported.
  CC:
      If no CC value is specified, clang is used.
  LD:
      If no LD value is specified, ${CROSS_COMPILE}ld is used.
  LLVM_IAS:
      If this is set to 1, the integrated assembler is used. No value means
      ${CROSS_COMPILE}as is used for assembling.
  NM:
      If no NM value is specified, the script will attempt to set NM to

>       llvm-nm, then ${CROSS_COMPILE}nm if they are found in PATH.

Reviewer: ```suggestion
      llvm-nm, then ${CROSS_COMPILE}nm if it is found in PATH.
```

  OBJCOPY:
      If no OBJCOPY value is specified, the script will attempt to set OBJCOPY

>       to llvm-objcopy, then ${CROSS_COMPILE}objcopy if they are found in PATH.

Reviewer: ```suggestion
      to llvm-objcopy, then ${CROSS_COMPILE}objcopy if it is found in PATH.
```

  OBJDUMP:
      If no OBJDUMP value is specified, the script will attempt to set OBJDUMP

>       to llvm-objdump, then ${CROSS_COMPILE}objdump if they are found in PATH.

Reviewer: ```suggestion
      to llvm-objdump, then ${CROSS_COMPILE}objdump if it is found in PATH.
```

  OBJSIZE:
      If no OBJSIZE value is specified, the script will attempt to set OBJSIZE

>       to llvm-size, then ${CROSS_COMPILE}size if they are found in PATH.

Reviewer: ```suggestion
      to llvm-size, then ${CROSS_COMPILE}size if it is found in PATH.
```

  REPO:
      Nicknames for trees:
        linux (default)
        linux-next
        5.4
        4.19
        4.14
        4.9
        4.4
        android-mainline
        android-5.4
        android-4.19
        android-4.14-stable
        android-4.9-q
  STRIP:
      If no STRIP value is specified, the script will attempt to set STRIP

>       to llvm-strip, then ${CROSS_COMPILE}strip if they are found in PATH.

Reviewer: ```suggestion
      to llvm-strip, then ${CROSS_COMPILE}strip if it is found in PATH.
```
